lib/galaxy/jobs/runners/drmaa.py

```diff
@@ -168,6 +180,49 @@ def queue_job( self, job_wrapper ):
         if native_spec:
             log.debug( "(%s) native specification is: %s", galaxy_id_tag, native_spec )
 
+        if(jt.nativeSpecification == None):
+            jt.nativeSpecification = '';
+        #Should transfer whole working directory when job is run on remote datasets
+        if(self.app.config.use_remote_datasets):
+            if(self.app.config.use_CCC_DRMAA):
+                jt.nativeSpecification += '\n' + 'TransferInput = '+job_wrapper.working_directory + os.sep;
+                jt.nativeSpecification += '\n' + 'ShouldTransferFiles = IF_NEEDED';
+            else:
+                jt.nativeSpecification += '\n' + 'transfer_input_files = '+job_wrapper.working_directory + os.sep;
+                jt.nativeSpecification += '\n' + 'should_transfer_files = YES';
+
+        #For CCC
+        if(self.app.config.use_CCC_DRMAA):
+            jt.nativeSpecification += '\noutput_aggregated=False\noutput_aggregation_type=merge';
+            #FIXME: CCC DRMAA code assumes all UUIDs are bounded by [], fix
+	    input_string_uuid_list = job_wrapper.get_input_string_uuids();
+	    if(len(input_string_uuid_list) == 0):
+	      jt.nativeSpecification += '\n' + 'input_CCC_DID_list=' + '[]'; 
+	    else:
+	      jt.nativeSpecification = jt.nativeSpecification + '\n' + 'input_CCC_DID_list=' + ','.join(map(lambda x:'[' + x + ']', 
+		input_string_uuid_list));
+	    output_string_uuid_list = job_wrapper.get_output_string_uuids();
+	    if(len(output_string_uuid_list) == 0 or job_wrapper.get_tool_id() == 'upload1'):
+	      jt.nativeSpecification += '\n' + 'output_CCC_DID_list=' + '[]';
+	    else:
+	      jt.nativeSpecification = jt.nativeSpecification + '\n' + 'output_CCC_DID_list=' + ','.join(map(lambda x:'[' + x + ']',
+		output_string_uuid_list));
+            jt.nativeSpecification = jt.nativeSpecification + '\n' + 'tool_id=' + job_wrapper.get_tool_id();
+            workflow_tuple = job_wrapper.get_workflow_invocation_info();
+            if(workflow_tuple):
+                (workflow_name, workflow_id, workflow_invocation_id) = workflow_tuple;
+                jt.nativeSpecification = jt.nativeSpecification + '\n' + 'workflow_name=' + workflow_name;
+                jt.nativeSpecification = jt.nativeSpecification + '\n' + 'workflow_id=' + str(workflow_id);
+                jt.nativeSpecification = jt.nativeSpecification + '\n' + 'workflow_invocation_id=' + str(workflow_invocation_id);
+            #Environment variables to pass to CCC
+            jt.nativeSpecification += '\nenvironment_vars=' + ','.join(self.CCC_environment_variables);
```

Comment:
Change to source env file in Galaxy
Move to dynamic destinations
Sub-class for CCC DRMAA